lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -315,211 +483,206 @@ runSelectionStep lens s
 -- Making change
 --------------------------------------------------------------------------------
 
+-- | Calculate change bundles from a set of selected inputs and outputs. Returns
+-- 'Nothing' if there are not enough 'Ada' inputs to satisfy minimum delta and
+-- minimum values in each token bundle. However, generate runtime errors if:
+--
+-- 1. The total input value is lesser than the total output value
+-- 2. The total output value is null
+--
+-- The pre-condition (1) should be satisfied by any result coming from
+-- `runSelection`. The pre-condition (2) is a undirect consequence of assigning
+-- a minimum UTxO value to every output token bundle.
 makeChange
-    :: NonEmpty TokenBundle
+    :: (TokenMap -> Coin)
+        -- A function which computes the minimum required Ada coins for a
+        -- particular output.
+    -> Coin
+        -- ^ The minimal (and optimal) delta between the total Ada inputs and
+        -- total Ada outputs. This typically captures fees plus key deposits.
+    -> Maybe Coin
+        -- ^ An extra source of Ada, if any.
+    -> NonEmpty TokenBundle
         -- ^ Token bundles of selected inputs
     -> NonEmpty TokenBundle
         -- ^ Token bundles of original outputs
-    -> NonEmpty TokenBundle
-        -- ^ Change bundles
```

Comment:
This is just a suggestion, by why not use the existing `MakeChangeData` type here (or some variation of it)?

**Advantages**:

1. It's arguably easier to read the function documentation if it explicitly refers to named fields such as `requiredCost` and `inputBundles` and so on (assuming someone is reading the generated Haddock docs).
2. We can reuse this type from the test suite, instead of having to define it there.
3. Sites that call `makeChange` will become a bit more readable, as instead of a long list of arguments, each argument will be named:

```hs
            mChangeGenerated = makeChange MakeChangeData
                { minCoinValue = minCoinValueFor
                , requiredCost = costFor s
                , extraCoinSource
                , inputBundles = view #tokens . snd <$> inputsSelected
                , outputBundles = view #tokens <$> outputsCovered
                }
```

Reply:
Indeed. I thought of it, and generally want to that for any function with more than 3 arguments. I wish records were easier to create on-the-fly in Haskell. I'll move the `MakeChangeData` from the test code to here.